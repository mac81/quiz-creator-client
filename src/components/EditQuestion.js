import React from 'react';
import { connect } from 'react-redux';

export class EditQuestion extends React.Component {
    render() {
        const { item } = this.props;

        if(!item) { return null }

        return (
            <div>
                <input defaultValue={item.question} />

                <table>
                    <thead>
                        <tr>
                            <th></th>
                            <th></th>
                            <th>Answer</th>
                            <th>Correct answer</th>
                        </tr>
                    </thead>
                    <tbody>
                    {item.answers.map((item, index) => (
                        <tr key={index}>
                            <td><button>Delete</button></td>
                            <td><button>Move</button></td>
                            <td><input defaultValue={item.answer}/></td>
                            <td><input type="radio" name="correctAnswer" checked={item.correctAnswer}/></td>
                        </tr>
                    ))}
                    </tbody>
                </table>
            </div>
        );
    }
}

const mapStateToProps = (state, props) => {
    return {
        item: state.questions.find(question => question._id === props.match.params.id)
    }
};

export default connect(mapStateToProps)(EditQuestion);