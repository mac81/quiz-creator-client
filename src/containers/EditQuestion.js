import React from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux'
import * as QuestionActions from 'actions/questions';
import * as AnswerActions from 'actions/answers';
import { SELECTORS as QUESTION_SELECTORS } from 'reducers/question';
import { SELECTORS as ANSWER_SELECTORS } from 'reducers/answers';

import {TextField, Paper} from 'material-ui';

import LabelIcon from 'material-ui/svg-icons/action/label-outline';
import DeleteIcon from 'material-ui/svg-icons/action/delete';

export class EditQuestion extends React.Component {

  constructor(props) {
    super(props);

    this.props.actions.loadQuestion(props.match.params.id, props.match.params.question_id);
    this.props.answerActions.loadAnswers(props.match.params.id, props.match.params.question_id);
  }

  componentWillReceiveProps(nextProps) {
    if(this.props.match.params.question_id !== nextProps.match.params.question_id) {
      this.props.actions.loadQuestion(this.props.match.params.id, nextProps.match.params.question_id);
    }
  }

  onUpdateQuestion = (e) => {
    this.props.actions.updateQuestion(e.target.name, e.target.value)
  }

  onDeleteQuestion = () => {
    this.props.actions.deleteQuestion(this.props.question._id);
  }

  onUpdateAnswerText = (e, answerId) => {
    //this.props.actions.updateQuestionAnswer('answerText', e.target.value, answerId)
  }

  onDeleteAnswer = (e) => {
    //this.props.actions.deleteQuestionAnswer(e.target.value);
  }

  onAddAnswer = () => {
    this.props.answerActions.createAnswer(this.props.match.params.id, this.props.match.params.question_id);
  }

  render() {
    const {question, answers} = this.props;

    if (!question) {
      return null
    }

    return (

        <Paper className="qc-panel">
          <div className="cq-panel-header">
            <LabelIcon className="cq-icon" color="#fff"/>
            <h4 className="cq-question-label">{question.label}</h4>
            <DeleteIcon onClick={this.onDeleteQuestion} color="#fff">Delete</DeleteIcon>
          </div>
          <div className="cq-panel-body">
          <TextField
            onChange={this.onUpdateQuestion}
            value={question.questionText}
            name="questionText"
            floatingLabelText="Question"
            fullWidth={true}
          />
        {/*<input defaultValue={question.questionText} name="questionText" onChange={this.onUpdateQuestion}/>*/}

        <table>
          <thead>
          <tr>
            <th></th>
            <th></th>
            <th>Answer</th>
            <th>Correct answer</th>
          </tr>
          </thead>
          <tbody>
          {/*{answers && answers.map((answer, index) => (*/}
            {/*<tr key={index}>*/}
              {/*<td>*/}
                {/*<button onClick={this.onDeleteAnswer} value={answer._id}>Delete</button>*/}
              {/*</td>*/}
              {/*<td>*/}
                {/*<button>Move</button>*/}
              {/*</td>*/}
              {/*<td>*/}
                {/*<input*/}
                  {/*defaultValue={answer.answerText}*/}
                  {/*name="answerText"*/}
                  {/*onChange={(e) => this.onUpdateAnswerText(e, answer._id)}*/}
                {/*/>*/}
              {/*</td>*/}
              {/*<td>*/}
                {/*<input*/}
                  {/*type="radio"*/}
                  {/*onChange={this.onUpdateQuestion}*/}
                  {/*name="correctAnswerId"*/}
                  {/*value={answer._id}*/}
                  {/*checked={answer._id === question.correctAnswerId}*/}
                {/*/>*/}
              {/*</td>*/}
            {/*</tr>*/}
          {/*))}*/}
          </tbody>
        </table>
            <button onClick={this.onAddAnswer}>Add answer</button>
          </div>
        </Paper>
    );
  }
}

const mapStateToProps = (state, props) => {
  return {
    question: QUESTION_SELECTORS.getQuestion(state),
    answers: ANSWER_SELECTORS.getAnswers(state)
  }
};

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(QuestionActions, dispatch),
  answerActions: bindActionCreators(AnswerActions, dispatch)
});

export default connect(mapStateToProps, mapDispatchToProps)(EditQuestion);