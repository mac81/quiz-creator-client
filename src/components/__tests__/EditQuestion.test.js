import React from 'react';
import renderer from 'react-test-renderer';

import { EditQuestion } from '../EditQuestion';

it('renders correctly', () => {
    const component = renderer.create(
        <EditQuestion item={{
            question: "Test",
            answers: [
                {
                    "answer": "Answer 1"
                }
            ]
        }}/>
    ).toJSON();
    expect(component).toMatchSnapshot();
});