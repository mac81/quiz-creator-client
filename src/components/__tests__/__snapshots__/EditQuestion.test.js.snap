// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    defaultValue="Test"
  />
  <table>
    <thead>
      <tr>
        <th />
        <th />
        <th>
          Answer
        </th>
        <th>
          Correct answer
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <button>
            Delete
          </button>
        </td>
        <td>
          <button>
            Move
          </button>
        </td>
        <td>
          <input
            defaultValue="Answer 1"
          />
        </td>
        <td>
          <input
            checked={undefined}
            name="correctAnswer"
            type="radio"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
